src/Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -96,157 +120,257 @@ public void Stop()
-                                Thread.Sleep(100);
                                 continue;
                             }
+                            else
+                            {
+                                this.logger.LogDebug("Process block from blockStore - '{0}'", this.chain.GetBlock(nextBlock.GetHash()));
+                                await this.ProcessAsync(nextBlock).ConfigureAwait(false);
+                            }
 
                             break;
                         }
-
-                        this.walletTip = next;
-                        this.walletManager.ProcessBlock(nextblock, next);
                     }
                 }
                 else
                 {
-                    ChainedHeader findTip = this.walletTip.FindAncestorOrSelf(newTip);
-                    if (findTip == null)
+                    this.logger.LogDebug("(-)[TIP_HEIGHT_<_WALLET_TIP]");
+
+                    // Check if any Blocks are queued up.
+                    while (this.blocksQueue.TryDequeue(out Block block))
                     {
-                        this.logger.LogTrace("(-)[NEW_TIP_BEHIND_NOT_IN_WALLET]");
-                        return;
+                        this.logger.LogTrace("Process block from blocksQueue - '{0}'", this.chain.GetBlock(block.GetHash()));
+                        await this.ProcessAsync(block).ConfigureAwait(false);
                     }
-
-                    this.logger.LogTrace("Wallet tip '{0}' is ahead or equal to the new tip '{1}'.", this.walletTip, newTip);
                 }
             }
-            else this.logger.LogTrace("New block follows the previously known block '{0}'.", this.walletTip);
-
-            this.walletTip = newTip;
-            this.walletManager.ProcessBlock(block, newTip);
+            catch (OperationCanceledException)
+            {
+                this.logger.LogInformation("Stopping WalletSynceManager...");
+            }
+            catch (Exception e)
+            {
+                this.logger.LogError("Exception occurred: {0}", e.ToString());
+            }
 
             this.logger.LogTrace("(-)");
         }
 
         /// <inheritdoc />
-        public virtual void ProcessTransaction(Transaction transaction)
+        private void ProducerBlock(ITargetBlock<Block> target, Block block)
         {
-            Guard.NotNull(transaction, nameof(transaction));
+            this.logger.LogDebug("()");
 
-            this.logger.LogTrace("({0}:'{1}')", nameof(transaction), transaction.GetHash());
-
-            this.walletManager.ProcessTransaction(transaction);
+            target.Post(block);
 
-            this.logger.LogTrace("(-)");
+            this.logger.LogDebug("(-)");
         }
 
         /// <inheritdoc />
-        public virtual void SyncFromDate(DateTime date)
+        private async Task ConsumerBlockAsync(ISourceBlock<Block> source, ConcurrentQueue<Block> queue)
```

Comment:
You should use `/// <inheritdoc />` when you don't want to rewrite the comments on a method implementing an interface member, which is not the case with these private methods.

Reply:
ok